.github/workflows/check_version.yml

```
@@ -0,0 +1,53 @@
+---
+name: "Check VERSION"
+
+on:
+  push:
+    branches:
+      - "main"
+      - "pre_commit_check"
+  pull_request_target: {}
+
+env:
+  OSS_VERSION_FILE: https://raw.githubusercontent.com/determined-ai/determined/main/VERSION
+
+permissions:
+  pull-requests: read
+  contents: read
+
+jobs:
+  compare-versions:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Get Determined OSS Version
+        run: |
+          # read determined OSS version file and get the current dev version
+          export OSS_VERSION="$(curl ${{ env.OSS_VERSION_FILE }} | grep -oE '^[0-9.]*')"
+          echo "OSS_VERSION=$OSS_VERSION" >> ${{ github.env }}
```

Probably better to use the github API since the version file contents are also the name of the latest release and the content of the tag for the latest release.  I doubt that the version file will ever move or that the contents will ever change, but the release naming convention is even less likely to change. :)

Likely the easiest thing here would be to use a third-party action and hide all the sausage-factory
```suggestion
      - name: Get Determined OSS Version
        id: oss
        uses: pozetroninc/github-action-get-latest-release@0.7.0
        with:
          owner: determined-ai
          repo: determined
          excludes: prerelease, draft
```

Then later, instead of using $OSS_VERSION, you'd use `${{ steps.oss.outputs.release }}

Slightly less preferable would be to use `gh api /repos/determined-ai/determined/releases/latest -q .name` instead of the curl to a blob, and store that output as OSS_VERSION.  I don't recall if you need a token for that or not - probably not, since it's a public repo.  If it fails, you just set a global variable to use our repo-wide CI token.